src/modules/aws_sns/mappers/topic.ts

```diff
@@ -0,0 +1,248 @@
+import isEqual from 'lodash.isequal';
+
+import {
+  SNS,
+  CreateTopicCommandInput,
+  paginateListTopics,
+  SetTopicAttributesCommandInput,
+} from '@aws-sdk/client-sns';
+import { parse as parseArn } from '@aws-sdk/util-arn-parser';
+
+import { AwsSnsModule } from '..';
+import { AWS, crudBuilder2, crudBuilderFormat, paginateBuilder } from '../../../services/aws_macros';
+import { Context, Crud2, MapperBase } from '../../interfaces';
+import { Topic } from '../entity';
+
+export class TopicMapper extends MapperBase<Topic> {
+  module: AwsSnsModule;
+  entity = Topic;
+  equals = (a: Topic, b: Topic) => {
+    return (
+      Object.is(a.arn, b.arn) &&
+      isEqual(a.contentBasedDeduplication, b.contentBasedDeduplication) &&
+      isEqual(a.deliveryPolicy, b.deliveryPolicy) &&
+      isEqual(a.displayName, b.displayName) &&
+      isEqual(a.fifoTopic, b.fifoTopic) &&
+      isEqual(a.kmsMasterKeyId, b.kmsMasterKeyId) &&
+      isEqual(a.policy, b.policy) &&
+      isEqual(a.signatureVersion, b.signatureVersion) &&
+      isEqual(a.tracingConfig, b.tracingConfig) &&
+      isEqual(a.dataProtectionPolicy, b.dataProtectionPolicy)
+    );
+  };
+
+  getTopicAttributes = crudBuilderFormat<SNS, 'getTopicAttributes', Record<string, string> | undefined>(
+    'getTopicAttributes',
+    arn => ({ TopicArn: arn }),
+    res => res?.Attributes,
+  );
+
+  getTopicDataProtection = crudBuilderFormat<SNS, 'getDataProtectionPolicy', string | undefined>(
+    'getDataProtectionPolicy',
+    arn => ({ ResourceArn: arn }),
+    res => res?.DataProtectionPolicy,
+  );
+
+  createTopic = crudBuilderFormat<SNS, 'createTopic', string | undefined>(
+    'createTopic',
+    input => input,
+    res => res?.TopicArn,
+  );
+
+  getTopics = paginateBuilder<SNS>(paginateListTopics, 'Topics', undefined, undefined);
+
+  setTopicAttributes = crudBuilder2<SNS, 'setTopicAttributes'>('setTopicAttributes', input => input);
+
+  putDataProtectionPolicy = crudBuilder2<SNS, 'putDataProtectionPolicy'>(
+    'putDataProtectionPolicy',
+    input => input,
+  );
+
+  deleteTopic = crudBuilder2<SNS, 'deleteTopic'>('deleteTopic', TopicArn => ({
+    TopicArn,
+  }));
+
+  attributeKeys = [
+    'contentBasedDeduplication',
+    'deliveryPolicy',
+    'displayName',
+    'kmsMasterKeyId',
+    'policy',
+    'signatureVersion',
+    'tracingConfig',
+  ];
+
+  capitalize = (s: string) => {
+    return s.charAt(0).toUpperCase() + s.slice(1);
+  };
+
+  async topicMapper(t: string, region: string, ctx: Context) {
```

Comment:
Can we rename this variable to something more explicit like `arn` or  `topicArn`?